File: basefiles_latest/src/main/java/com/rp/basefiles/BasePresenterVM.java
package com.rp.basefiles;

import android.arch.lifecycle.ViewModel;

/**
 * Created by rahul on 4/1/18.
 */

public abstract class BasePresenterVM<V extends IBaseView> extends ViewModel implements IBasePresenter<V> {

    private V view;

    @Override
    public void onAttach(V baseView) {
        this.view = baseView;
    }

    public V view() {
        return view;
    }

    @Override
    public void onDestroy() {
        onCleared();
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        view = null;
    }
}

File: SampleApp/basefiles/src/main/java/com/rp/basefiles/BasePresenterVM.kt
package com.rp.basefiles

import androidx.lifecycle.ViewModel

abstract class BasePresenterVM<V : IBaseView> : ViewModel(), IBasePresenter<V> {

    private var view: V? = null

    override fun onAttach(baseView: V) {
        this.view = baseView
    }

    fun view(): V? = view

    override fun onDestroy() = onCleared()

    override fun onCleared() {
        super.onCleared()
        view = null
    }
}